mantidproject/mantid
# Fix SANS 2d reduction selection ignored

**Description of work.**
Fixes SANS 2D reduction modes being ignored when the user provides a custom user file

**Report to:** ISIS/Steve

**To test:**
- Download the ISIS Training data set
- Extract and check you have `loqdemo`
- Open the ISIS SANS GUI
- Set your user file to `MaskFile.txt`
- Set your batch file to `batch_mode_reduction.csv`
- Edit the table so that there is only one run, in the blank user file tab enter `MaskFile.txt`
- Hit process all, the processed dataset should be 1D (I plot to check)
- Change it to 2D and hit process all, the processed dataset should be 2D (colour fill plot)


Fixes #27300 .

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
